org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/SnippetCompletionProposal.java

```diff
@@ -336,13 +338,30 @@ private static List<CompletionItem> getGenericSnippets(SnippetCompletionContext
 		Template[] availableTemplates = Arrays.stream(templates).filter(t -> javaContext.canEvaluate(t)).toArray(Template[]::new);
 		Template[] availableTemplatesAll = Arrays.stream(templatesAll).filter(t -> javaContextAll.canEvaluate(t)).toArray(Template[]::new);
 		List<CompletionProposal> proposals = new ArrayList<>();
+		int indexReduction = 0;
 		for (int i = 0; i < availableTemplates.length + availableTemplatesAll.length; i++) {
 			Template template;
 			if (i < availableTemplates.length) {
 				template = availableTemplates[i];
 			} else {
 				template = availableTemplatesAll[i - availableTemplates.length];
 			}
+
+			IJavaElement enclosingElement = completionContext.getEnclosingElement();
+			if (enclosingElement.getElementType() == IJavaElement.TYPE && ((IType) enclosingElement).isInterface()) {
+				if (template.getName().equals("method")) {
+					Template newTemplate = new Template(template.getName(), template.getDescription(), template.getContextTypeId(), INTERFACE_METHOD_SNIPPET, false);
+					template = newTemplate;
+				}
+				else if (template.getName().equals("ctor")) {
```

Comment:
Reverse the comparison and use `CodeSnippetTemplate.CTOR.name()` .

It's just better practice to compare against the term that is least likely to be `null`